File: 프로그래머스/lv1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.java
import java.util.ArrayList;
import java.util.Comparator;

class Solution {
    public int[] solution(int[] arr, int divisor) {
        ArrayList list = new ArrayList();

        for(int a: arr) {
            if(a % divisor == 0) {
                list.add(a);
            }
        }

        if(list.size() == 0) {
            list.add(-1);
        }

        list.sort(Comparator.naturalOrder());

        return list.stream().mapToInt(x -> (int)x).toArray();
    }
}

File: 프로그래머스/1/12910. 나누어 떨어지는 숫자 배열/나누어 떨어지는 숫자 배열.go
import (
    "sort"
)

func solution(arr []int, divisor int) []int {
    sort.Slice(arr, func(i, j int) bool {
        return arr[i] < arr[j]
    })    
    
    var answer []int
    
    for _, v := range arr {
        if v % divisor == 0 {
            answer = append(answer, v)
        }
    }
    
    if len(answer) == 0 {
        answer = append(answer, -1)
    }
    
    return answer
}